torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -998,6 +998,59 @@ void TensorExprKernel::LowerToBackend(BackendType backend_type) {
             "Invalid loop-level: " + std::to_string(loop_levels));
       }
     }
+  } else if (backend_type == kLLVMCodeGen) {
+    l.ApplyInlines();
+
+    std::vector<For*> inner_loops;
+    std::vector<For*> worklist;
+
+    if (For* root_f = dynamic_cast<For*>(l.root_stmt())) {
+      worklist.push_back(root_f);
```

Comment:
I think `root_stmt` is always a `Block`

Reply:
Nope!  I wrote the loop below because I hit cases where it wasn't.